spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -691,6 +691,10 @@ else if (methodName.equals("isTransactional")) {
 			}
 			try {
 				if (this.target == null || !this.target.isOpen()) {
+					if (this.target instanceof PublisherCallbackChannel) {
+						this.target.close();
+						throw new InvocationTargetException(new AmqpException("PublisherCallBackChannel is closed"));
```

Comment:
Seems for me nothing to complain about. Only the fix for this exception message when there is no such a `PublisherCallBackChannel` class :smile: